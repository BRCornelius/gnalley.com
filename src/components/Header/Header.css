.nav-menu {
    background-color: #DADADA;
    border: 1px solid #000035;
    display: inline-flex;
    width: 100%;
}

.blank-bar {
    height: 6rem;
    width: 100vw;
}

.hamburger-menu {
    background-color: #DADADA;
    padding: 1rem;
    border-radius: 50%;
    border: 1px solid #000035;
    opacity: 0.5;
    position: fixed;
    top: 1rem;
    right: 1rem;
    z-index: 9999;
}

.hamburger-open {
    background-color: #DADADA;
    padding: 1rem;
    border-radius: 5%;
    position: fixed;
    opacity: 0.8;
    top: 1rem;
    right: 1rem;
    z-index: 9999;
}

.nav-link-container {
    display: inline-flex;
    justify-content: space-between;
    padding: 0.5rem 3rem;
    width: 100%;
  }
  
  .nav-link {
    color: #000035;
    text-decoration: none;
  }
  

.logo {
    padding: 0.5rem;
}
  