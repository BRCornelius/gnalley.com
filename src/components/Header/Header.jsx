import React, { useRef, useState } from 'react';
import { Link } from 'react-router-dom';
import { isMobile, useOutsideClick } from '../../utils';
import './Header.css';

export const Header = () => {
    const [open, setOpen] =useState(false)
    const toggleNav = () => setOpen(!open)
    const wrapperRef = useRef(null)

    useOutsideClick(wrapperRef, toggleNav)

    return isMobile
        ? <>
            <div className="blank-bar">
                <img alt="logo" className="logo" src="http://www.gnalley.com/wp-content/uploads/2014/08/LogoTaglineBlue.png" />
            </div>
            <div className={open ? "hamburger-open" : "hamburger-menu"}>
                {!open && <img alt="menu" src="https://img.icons8.com/material-outlined/24/000000/menu--v1.png" onClick={toggleNav}/>}
                {open && <div className="links-menu" ref={wrapperRef}>
                    <Link to="/" className="nav-link" onClick={toggleNav}><h1>Home</h1></Link>
                    <Link to="/firm" className="nav-link" onClick={toggleNav}><h1>Firm</h1></Link>
                    <Link to="/team" className="nav-link" onClick={toggleNav}><h1>Team</h1></Link>
                    <Link to="/contact" className="nav-link" onClick={toggleNav}><h1>Contact</h1></Link>
                </div>}
            </div>
        </>
        : <div className="nav-menu">
            <img alt="logo" className="logo" src="http://www.gnalley.com/wp-content/uploads/2014/08/LogoTaglineBlue.png" />
            <nav className="nav-link-container">
                <Link to="/" className="nav-link"><h1>Home</h1></Link>
                <Link to="/firm" className="nav-link"><h1>Firm</h1></Link>
                <Link to="/team" className="nav-link"><h1>Team</h1></Link>
                <Link to="/contact" className="nav-link"><h1>Contact</h1></Link>
            </nav>
        </div>
}