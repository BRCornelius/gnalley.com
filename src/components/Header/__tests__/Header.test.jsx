import { createMemoryHistory } from 'history';
import { fireEvent, render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { Header } from '../Header';
import { Route, Switch } from 'react-router-dom';

const MockSiteNavigation = () => <Switch>
    <Route path="/contact">
        <div data-testid="contact-page" />
    </Route>
    <Route path="/team">
        <div data-testid="team-page" />
    </Route>
    <Route path="/firm">
        <div data-testid="firm-page" />
    </Route>
    <Route path="/">
        <div data-testid="splash-page" />
    </Route>
</Switch>

test('Full app navigation', () => {
    const history = createMemoryHistory();
    const { getByTestId, getByText, getByAltText } = render(
        <Router history={history}>
            <Header />
            <MockSiteNavigation />
        </Router>
    );
    expect(getByTestId('splash-page')).toBeInTheDocument();
    fireEvent.click(getByAltText('menu'));
    fireEvent.click(getByText('Firm'));
    expect(getByTestId('firm-page')).toBeInTheDocument();
})

test('Bad navigation routes to homepage', () => {
    const history = createMemoryHistory();
    history.push('/bad-route')
    const { getByTestId } = render(
        <Router history={history}>
            <Header />
            <MockSiteNavigation />
        </Router>
    );
    expect(getByTestId('splash-page')).toBeInTheDocument();
})

test('Renders correctly', () => {
    const { container, getByAltText } = render(<Header />)
    expect(getByAltText('logo')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})