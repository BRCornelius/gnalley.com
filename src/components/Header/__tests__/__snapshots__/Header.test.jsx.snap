// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <div
    class="blank-bar"
  >
    <img
      alt="logo"
      class="logo"
      src="http://www.gnalley.com/wp-content/uploads/2014/08/LogoTaglineBlue.png"
    />
  </div>
  <div
    class="hamburger-menu"
  >
    <img
      alt="menu"
      src="https://img.icons8.com/material-outlined/24/000000/menu--v1.png"
    />
  </div>
</div>
`;
